src/importlinter/application/contract_utils.py

```diff
@@ -33,16 +32,35 @@ def remove_ignored_imports(
     Returns:
         A list of any warnings to be surfaced to the user.
     """
-    imports, unresolved_expressions = helpers.resolve_import_expressions(
-        graph=graph, expressions=ignore_imports if ignore_imports else []
-    )
+    imports_to_remove = set()
+    unresolved_expressions = set()
+    for import_expression in ignore_imports or []:
+        matched_imports = graph.find_matching_direct_imports(
+            import_expression=str(import_expression)
+        )
+        if matched_imports:
+            imports_to_remove.update(
+                {
+                    DirectImport(
+                        importer=Module(matched_import["importer"]),
+                        imported=Module(matched_import["imported"]),
+                    )
+                    for matched_import in matched_imports
+                }
+            )
+        else:
+            unresolved_expressions.add(import_expression)
 
     warnings = _handle_unresolved_import_expressions(
         unresolved_expressions,
         unmatched_alerting,
     )
 
-    helpers.pop_imports(graph, imports)
+    for import_to_remove in imports_to_remove:
+        graph.remove_import(
+            importer=import_to_remove.importer.name,
+            imported=import_to_remove.imported.name,
+        )
```

Comment:
I'm not using `helpers.pop_imports` here, since: 
* `helpers.pop_imports` will call `get_import_details` for every import, which will have a significant performance overhead (we would have to cross the python-rust interface for every call).
* `helpers.pop_imports` will raise `MissingImport` is no import details (line number + contents) exist. This seems a bit wrong to me - in grimp it's possible for an import to exist in the graph without import details?